src/scales/scale.time.js

```diff
@@ -524,7 +524,7 @@ module.exports = Scale.extend({
 		var datasets = [];
 		var labels = [];
 		var i, j, ilen, jlen, data, timestamp;
-		var dataLabels = chart.data.labels || [];
+		var dataLabels = options.labels || chart.data.labels || [];
```

Comment:
Instead, shouldn't we make this feature consistent across all scales and move [this method](/chartjs/Chart.js/blob/master/src/scales/scale.category.js#L15) in `core.scale.js`?

```javascript
/**
 * @private
 */
_getLabels() {
    var data = this.chart.data;
    return this.options.labels || (this.isHorizontal() ? data.xLabels : data.yLabels) || data.labels; 
}
```

And since this method is commented [`@private` since the beginning](https://github.com/chartjs/Chart.js/pull/2849/files#diff-be0d9a8d9773b878dd378bd35a5d81ddR15), I would rename to `_getLabels`.

Reply:
Done